### PR TITLE
O365 - Search and Delete Last Week - attempt 2

### PR DESCRIPTION
**Pull request by ali-sawyer**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
n/a

## Description
This is a follow-up to: https://github.com/demisto/content/pull/33858. I was having issues pushing to the existing PR, perhaps because of how it was created with the conversion from the Marketplace UI contribution process to GitHub, so I created a new one.

This change is to implement the `SearchThisWeek` input for O365, to limit the search to the last 7 days by default. Currently this input is ignored and not used for O365 (it is only used for Gmail) so this is a _breaking change_.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by anas-yousef** [comment]
Hi @ali-sawyer
Thank you for your contribution.
Your PR review might be slightly delayed due to the upcoming holiday, I apologize in advance. The holiday will be over by April 30th.
I'm available for any of your questions here or on slack.
Thanks again.

**Review by ssokolovich — COMMENTED** [review]
Awesome - great work! 
A couple of notes from my end:
* run the `demisto-sdk format` on this file - so it will be reorganized and formatted according to our standard.
* Please also update the Playbook image under the ../docs/ path.
* Add additional condition that validates the KQL doesn't contain the `LastModifiedTime` parameter before we are adding the "Last 7 days" time frame.
* Not sure if you saw that, but the lines look a bit messy which makes the flow a bit hard to understand.  Adding an image below.
image

Please ping me once you are ready - would like to see a quick demo from your env that the logic works :)

File: Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml [hunk 1]
```diff
@@ -1114,8 +1299,5 @@ inputs:
   playbookInputQuery:
 outputs: []
 tests:
-- No tests (auto formatted)
+- Send Investigation Summary Reports - Test
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml:1302]: please remove this test - it was added probably by the SDK.

File: Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml [hunk 2]
```diff
@@ -39,19 +35,20 @@ tasks:
...
+    continueonerrortype: ""
   "1":
     id: "1"
     taskid: 0a7ca589-107b-48e4-8705-2becca697165
     type: playbook
     task:
       id: 0a7ca589-107b-48e4-8705-2becca697165
       version: -1
-      name: Search And Delete Emails - EWS
+      name: search_and_delete_emails_-_ews
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml:46]: This task name was changed for some reason - please revert it.

File: Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml [hunk 3]
```diff
@@ -39,19 +35,20 @@ tasks:
...
       description: This playbook searches EWS to identify and delete emails with similar attributes of a malicious email.
-      playbookName: Search And Delete Emails - EWS
       type: playbook
       iscommand: false
       brand: ""
+      playbookId: search_and_delete_emails_-_ews
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml:51]: The playbook id here was also changed - maybe it's due to custom content you played with on your env?

File: Packs/CommonPlaybooks/doc_files/Search_And_Delete_Emails_-_Generic_-_v2_6_0.png
Comment (ssokolovich) [inline Packs/CommonPlaybooks/doc_files/Search_And_Delete_Emails_-_Generic_-_v2_6_0.png:1]: Please add an image with a white background (and not the dark mode... it's easier to read the tasks names and look at it)

**Comment by ssokolovich** [comment]
Looks Good @ali-sawyer, seems like the format indeed worked. 
Once you made the latest changes we can schedule a demo :)

File: Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml [hunk 4]
```diff
@@ -118,12 +116,13 @@ tasks:
...
     isautoswitchedtoquietmode: false
+    continueonerrortype: ""
   "3":
     id: "3"
-    taskid: b17bdcb8-c603-448c-8fd6-77bca91e7e2f
+    taskid: 86fc3213-5a75-41dc-8c9b-b924335c28a1
     type: playbook
     task:
-      id: b17bdcb8-c603-448c-8fd6-77bca91e7e2f
+      id: 86fc3213-5a75-41dc-8c9b-b924335c28a1
```
Comment (ali-sawyer) [inline Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml:125]: @ssokolovich should we be concerned that these taskids changed?

File: Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml [hunk 5]
```diff
@@ -1295,9 +1373,9 @@ inputs:
   value:
     simple: "true"
   required: false
-  description: Whether to limit the search to the current week.
+  description: Whether to limit the search to the current week. Note that if you are using the LastModifiedTime filter in the O365KQL input, this input will not be taken into consideration.
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml:1376]: this is the generic playbook - so I think that better to keep this note only for the `O365KQL` input description.

File: Packs/CommonPlaybooks/doc_files/Search_And_Delete_Emails_-_Generic_-_v2_6_0.png
Comment (ssokolovich) [inline Packs/CommonPlaybooks/doc_files/Search_And_Delete_Emails_-_Generic_-_v2_6_0.png:1]: Can we re-arrange it to the way it was? some of the arrows got messed up and might confuse our users.

File: Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml [hunk 6]
```diff
@@ -1030,19 +1044,268 @@ tasks:
...
+      value:
+        simple: ${inputs.O365KQL}
+    separatecontext: false
+    skipunavailable: false
+    task:
+      brand: ""
+      description: Set a value in context under the key you entered.
+      id: 488ba512-437e-4bc1-81f8-0ef84aa353ae
+      iscommand: false
+      name: Create KQL query - search all time
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_1.yml:1202]:
```suggestion
      name: Set the non-modified KQL query
```

**Review by ssokolovich — COMMENTED** [review]
Looks good - 3 more notes from my side and we are good to go!

**Comment by ali-sawyer** [comment]
@ssokolovich Requested changes made.

**Review by ssokolovich — APPROVED** [review]

**Pull request merged**